src/cpp/astar.cpp

```diff
@@ -73,6 +117,19 @@ static PyObject *astar(PyObject *self, PyObject *args) {
   nodes_to_visit.push(start_node);
 
   int* nbrs = new int[8];
+  
+  heuristic_ptr heuristic_func;
+
+  // Get the heuristic method to use
+  if (heuristic_override == DEFAULT) {
+    if (diag_ok) {
+      heuristic_func = linf_norm;
+    } else {
+      heuristic_func = l1_norm;
+    }
+  } else {
+    heuristic_func = select_heuristic(heuristic_override);
+  }
```

Comment:
Is this correct? If someone chooses to allow diagonal moves, then they cannot also choose to use the Manhattan distance. Diagonal moves would violate the admissibility of this heuristic. Or am I misunderstanding something?

Reply:
I think the point was to allow people to pick any heuristic that is available. I can remove the two default if that is the direction you want to go. Which I knew you would point out :)

Reply:
Yes, thanks for fixing this. I don't want people to choose heuristics that lead to sub-optimal paths.